src/install/bin.zig

```diff
@@ -311,38 +312,127 @@ pub const Bin = extern struct {
+
+                std.debug.assert(strings.hasPrefixComptime(target_path, "..\\"));
 
-                    setPermissions(node_modules.fd, dest_path);
-                    var target_path_trim = target_path;
-                    if (strings.hasPrefix(target_path_trim, "../")) {
-                        target_path_trim = target_path_trim[3..];
+                const target_wpath = bun.strings.toWPathNormalized(&filename1_buf, target_path[3..]);
+                var destination_wpath: []u16 = bun.strings.convertUTF8toUTF16InBuffer(&filename2_buf, dest_path);
+
+                const first_content_chunk = contents: {
+                    const fd = bun.sys.openatWindows(
+                        this.package_installed_node_modules,
+                        target_wpath,
+                        std.os.O.RDONLY,
+                    ).unwrap() catch |err| {
+                        this.err = err;
+                        return;
+                    };
+                    defer _ = bun.sys.close(fd);
+                    const reader = fd.asFile().reader();
+                    const read = reader.read(&read_in_buf) catch |err| {
+                        this.err = err;
+                        return;
+                    };
+                    if (read == 0) {
+                        this.err = error.FileNotFound;
+                        return;
                     }
-                    setPermissions(node_modules.fd, target_path_trim);
+                    break :contents read_in_buf[0..read];
+                };
+
+                const shim = WinBinLinkingShim{
+                    .bin_path = target_wpath,
+                    .shebang = WinBinLinkingShim.Shebang.parse(first_content_chunk, target_wpath) catch {
+                        this.err = error.InvalidBinContent;
+                        return;
+                    },
+                };
+
+                const len = shim.encodedLength();
+                if (len > shim_buf.len) {
+                    this.err = error.InvalidBinContent;
                     return;
                 }
+                const metadata = shim_buf[0..len];
+                shim.encodeInto(metadata) catch {
+                    this.err = error.InvalidBinContent;
+                    return;
+                };
+
+                destination_wpath.len += 5;
+                @memcpy(destination_wpath[destination_wpath.len - 5 ..], &[_]u16{ '.', 'b', 'u', 'n', 'x' });
+
+                if (node_modules.createFileW(destination_wpath, .{
+                    .truncate = true,
+                })) |file| {
+                    defer file.close();
+                    file.writer().writeAll(metadata) catch |err| {
+                        this.err = err;
+                        return;
+                    };
+                } else |err| {
+                    if (err != error.PathAlreadyExists) {
+                        this.err = err;
+                        return;
+                    }
```

Comment:
if the path already exists, we need to check if the bin path encoded into the existing `.bunx` file is still valid and replace it if not (fixed on other platforms earlier today #8293)